src/githubcity/ghcity.py

```diff
@@ -527,7 +527,9 @@ def __readAPI(self, url):
                     log_message += str(reset - now_sec)
                     log_message += " secs"
                     self.__logger.warning(log_message)
-                    time.sleep(reset - now_sec)
+                    sleep_duration = reset - now_sec
+                    if sleep_duration > 0:
```

Comment:
It is not a good idea to not sleep. Usually, this time is negative because the server is sending a reply with code error 403 and `reset` is negative. It would be better to wait some time (30 seconds, for example).